File: apps_task/C1_Blog_Cake/cakephp/app/Controller/BlogsController.php
<?php
	class BlogsController extends AppController{
		public $helpers = array('Html', 'Form');
		public $components = array('Session');

		public function index(){
			$loggedin=0;
			$this->set('uid',$this->Auth->user('id'));
			if($this->Auth->user('id') == null){
				$loggedin = 0;
			}else{
				$loggedin = 1;
			}
			$this->set('loggedin', $loggedin);

			$blogs = $this->Blog->find('all');

			foreach ($blogs as &$blog){
				$blog['Blog']['user_name'] = $blog['User']['username'];
			}

			$this->set('blogs',$blogs);
		}


		public function view($id){

			$blog = $this->Blog->findById($id);
			$posts = $this->Blog->Post->findAllByBlog_id($id);

			foreach ($posts as &$post){
				$post['Post']['comment_count'] = $this->Blog->Post->Comment->find('count' , array('conditions' => array('Comment.post_id'=> $post['Post']['id'])));
			}
			
			$this->set('posts' , $posts);
			$this->set('blog' , $blog);
		}

		public function add() {
	
        	if ($this->request->is('post')) {
            	$this->Blog->create();
            	$this->Blog->set('user_id', $this->Auth->user("id"));
            if ($this->Blog->save($this->request->data)) {
                $this->Session->setFlash('Your blog has been saved.');
                $this->redirect(array('action' => 'index'));
            } else {
                $this->Session->setFlash('Unable to add your blog.');
            }
        }
    }
	
	}
?>

File: apps_task/C4_Blog_Rails/BlogApp/app/controllers/blogs_controller.rb
class BlogsController < ApplicationController

  def index
    @blogs = Blog.all
  end

  def show
    @blog = Blog.find(params[:id])
    @posts = Post.find(:all, :conditions => { :blog_id => @blog.id })
    @search = Post.search do 
      fulltext params[:search]
    end
      @posts = @search.results
    respond_to do |format|
      format.html # show.html.erb
      format.json { render json: @post }
    
  end
  end 


	def new
	@blog = Blog.new
	end

	def create
    @blog = Blog.new(params[:blog])
    @blog.update_attributes(:user_id => session[:user_id])
    if @blog.save
      redirect_to root_url, notice: "Your blog has been created!"
    else
      render "new"
    end
  end
  
  def edit
    @blog = Blog.find(:first, :conditions => { :user_id => session[:user_id] })
  end

  def update
    @blog = Blog.find(:first, :conditions => { :user_id => session[:user_id] })
    @blog.update_attributes(params[:blog])
    if @blog.save
      redirect_to root_url, notice: "Your blog has been renamed!"
    else
      render "edit"
    end
  end
end
